src/redux/features/plantings/plantingsSelectors.ts

```diff
@@ -81,19 +81,23 @@ export const selectPlantingProgress = createSelector(
 // selector to search plantings
 export const searchPlantingProgress = createSelector(
   [
-    (state: RootState, query: string, plantingCompleted?: boolean) => selectPlantingProgress(state),
-    (state: RootState, query: string, plantingCompleted?: boolean) => query,
-    (state: RootState, query: string, plantingCompleted?: boolean) => plantingCompleted,
+    (state: RootState, query: string, plantingCompleted?: boolean, plantingSiteId?: number) =>
+      selectPlantingProgress(state),
+    (state: RootState, query: string, plantingCompleted?: boolean, plantingSiteId?: number) => query,
+    (state: RootState, query: string, plantingCompleted?: boolean, plantingSiteId?: number) => plantingCompleted,
+    (state: RootState, query: string, plantingCompleted?: boolean, plantingSiteId?: number) => plantingSiteId,
   ],
-  (plantingProgress, query, plantingCompleted) => {
+  (plantingProgress, query, plantingCompleted, plantingSiteId) => {
     return plantingProgress?.reduce((acc, curr) => {
       const { siteId, siteName, totalPlants, reported } = curr;
       if (reported && reported.length > 0) {
         reported?.forEach((progress) => {
           const matchesQuery = !query || regexMatch(progress.subzoneName, query);
           const matchesPlantingCompleted =
             plantingCompleted === undefined || progress.plantingCompleted === plantingCompleted;
-          if (matchesQuery && matchesPlantingCompleted) {
+          const matchesPlantingSite =
+            plantingSiteId === undefined || plantingSiteId === -1 || siteId === plantingSiteId;
```

Comment:
nice.. are siteId and plantingSiteId both numbers or do we need a toString() on them for comparison?

Reply:
we need to add checks in line 104 as well, this is for sites that don't have zones, right now the site drop down doesn't filter out sites without zones.

Reply:
Thanks, will update it

Reply:
> nice.. are siteId and plantingSiteId both numbers or do we need a toString() on them for comparison?

They are both numbers